nutanix/resource_nutanix_user_groups.go

```diff
@@ -0,0 +1,275 @@
+							Required: true,
+						},
+					},
+				},
+			},
+			"directory_service_ou": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				ConflictsWith: []string{"directory_service_user_group", "saml_user_group"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"distinguished_name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"metadata": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"project_reference": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"categories": categoriesSchema(),
+			"owner_reference": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceNutanixUserGroupsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).API
+
+	request := &v3.UserGroupIntentInput{}
+	spec := &v3.UserGroupSpec{}
+	res := &v3.UserGroupResources{}
+	metadata := &v3.Metadata{}
+
+	if err := getMetadataAttributes(d, metadata, "user_group"); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if ds, ok := d.GetOk("directory_service_user_group"); ok {
+		res.DirectoryServiceUserGroup = expandDirectoryUserGroup(ds.([]interface{}))
+	}
+
+	if ds, ok := d.GetOk("directory_service_ou"); ok {
+		res.DirectoryServiceOU = expandDirectoryUserGroup(ds.([]interface{}))
+	}
+
+	if su, ok := d.GetOk("saml_user_group"); ok {
+		res.SamlUserGroup = expandSamlUserGroup(su.([]interface{}))
+	}
+
+	spec.Resources = res
+	request.Spec = spec
+	request.Metadata = metadata
+
+	// Create User Group API
+
+	resp, err := conn.V3.CreateUserGroup(ctx, request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	uuid := *resp.Metadata.UUID
+	taskUUID := resp.Status.ExecutionContext.TaskUUID.(string)
+
+	// Wait for the UserGroup to be available
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"PENDING", "RUNNING"},
+		Target:     []string{"SUCCEEDED"},
+		Refresh:    taskStateRefreshFunc(conn, taskUUID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      userGroupDelayTime,
+		MinTimeout: userGroupMinTimeout,
+	}
+
+	if _, errWaitTask := stateConf.WaitForStateContext(ctx); errWaitTask != nil {
+		return diag.Errorf("error waiting for user group (%s) to create: %s", uuid, errWaitTask)
+	}
+
+	d.SetId(uuid)
+	return resourceNutanixUserGroupsRead(ctx, d, meta)
+}
+
+func resourceNutanixUserGroupsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).API
+
+	resp, err := conn.V3.GetUserGroup(d.Id())
+	if err != nil {
+		if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
+			d.SetId("")
+			return nil
```

Comment:
If entity is not found, we can return the error right ? Instead of nil.

Reply:
Done.